client/js/lounge.js

```diff
@@ -465,7 +471,35 @@ $(function() {
 		return false;
 	});
 
+	sidebar.on("click", ".add-channel", (e) => {
+		e.preventDefault();
+		const id = $(e.target).data("id");
+		const joinForm = $(`#join_channel_${id}`);
+		joinForm.toggle();
+	});
+
+	sidebar.on("submit", ".join_form", function(e) {
+		e.preventDefault();
+		const form = $(this);
+		const channel = form.find("input[name='channel']").val();
+		const key = form.find("input[name='key']").val();
+		const chan = utils.findCurrentNetworkChan(channel);
+		if (chan.length) {
+			chan.click();
+		} else {
+			socket.emit("input", {
+				text: `/join ${channel} ${key}`,
+				target: $("#chat").data("id"),
+			});
+		}
+		form.toggle();
```

Comment:
Use `hide` instead.

Reply:
When you press the add button multiple times it feels natural for it to toggle it imo

Reply:
This is submit handler, not the `+` button click.